deepchem/models/keras_model.py

```diff
@@ -432,8 +465,28 @@ def fit_generator(self,
         c(self, current_step)
       if self.tensorboard and should_log:
         self._log_scalar_to_tensorboard('loss', batch_loss, current_step)
-      if self.wandb and should_log:
-        wandb.log({'loss': batch_loss}, step=current_step)
+      if self.wandb_logger is not None:
+
+        #Get epoch number
+        dataset_size = len(self.wandb_logger.datasets[self.wandb_logger.train_dataset_name])
+        batches_per_epoch = math.ceil(dataset_size / self.batch_size)
+        epoch_num = current_step // batches_per_epoch
+
+        #Log loss and metrics at specified interval
+        if should_log:
+          all_metrics = self.wandb_logger.calculate_metrics(self)
+          data = dict({'train/loss': batch_loss,
+                       'epoch': epoch_num})
+          data.update(all_metrics)
+          self.wandb_logger.log_data(data, step=current_step)
+        else:
+          #otherwise just log epoch
+          self.wandb_logger.log_data({'epoch': epoch_num}, step=current_step)
+
+
+    #Close WandbLogger
+    if self.wandb_logger is not None:
+      self.wandb_logger.finish()
```

Comment:
Just a reminder that we'll have to make sure that any model saving to Artifacts happens before this call